src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMIT.java

```diff
@@ -66,6 +69,43 @@ public void testCheckout() throws Exception {
         assertTrue(build.getWorkspace().child("add_file").isDirectory());
     }
 
+    @Test
+    public void testCheckoutAndPush() throws Exception {
+        project.setScm(createSCM("*/master"));
+        String uniqueMessage = UUID.randomUUID().toString();
+        Shell postScript = new Shell("git checkout master\n" +
+                                     "echo \"Hello, World!\" >> test.txt\n" +
+                                     "git add test.txt\n" +
+                                     "git commit -m \"" + uniqueMessage + "\"\n" +
+                                     "git push");
+
+        project.getBuildersList().add(postScript);
+        project.save();
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        BitbucketRevisionAction revisionAction = build.getAction(BitbucketRevisionAction.class);
+        String restPath = new StringBuilder().append(BitbucketUtils.BITBUCKET_BASE_URL)
+                .append("/rest/api/1.0/projects/")
+                .append(BitbucketUtils.PROJECT_KEY)
+                .append("/repos/")
+                .append(BitbucketUtils.REPO_SLUG)
+                .append("/commits?since=")
+                .append(revisionAction.getRevisionSha1())
+                .toString();
+
+        assertEquals(SUCCESS, build.getResult());
+        RestAssured
+                .given()
+                    .auth().preemptive().basic(BitbucketUtils.BITBUCKET_ADMIN_USERNAME, BitbucketUtils.BITBUCKET_ADMIN_PASSWORD)
+                .expect()
+                    .statusCode(200)
+                    .body("values.size", equalTo(1))
+                    .body("values[0].message", equalTo(uniqueMessage))
```

Comment:
This is kind of a hack. The more appropriate method would be getting the commit hash, but this would require parsing the logs (as it's done remotely from jenkins), so this was a compromise.

Reply:
Don't think this is a hack at all. Seems good to me